Compiler/src/typeinfer.jl

```diff
@@ -1280,8 +1280,20 @@ function typeinf_ext_toplevel(methods::Vector{Any}, worlds::Vector{UInt}, trim::
                     ci isa CodeInstance && !use_const_api(ci) && push!(tocompile, ci)
                 end
             elseif item isa SimpleVector
-                push!(codeinfos, item[1]::Type)
-                push!(codeinfos, item[2]::Type)
+                (rt::Type, sig::Type) = item
+                # make a best-effort attempt to enqueue the relevant code for the ccallable
+                ptr = ccall(:jl_get_specialization1,
+                            #= MethodInstance =# Ptr{Cvoid}, (Any, Csize_t, Cint),
+                            sig, get_world_counter(), #= mt_cache =# 0)
```

Comment:
We run this code for every world requested to be included in the output
```suggestion
                            sig, this_world, #= mt_cache =# 0)
```